File: HeartDataMiniProgram/pages/report/report.js
// pages/report/report.js
Page({

  /**
   * 页面的初始数据
   */
  data: {
    reports: {}
  },
  /**
   * 生命周期函数--监听页面加载
   */
  onLoad: function (options) {
    var that = this;
    // 解析JSON对象
    /*
    {
      "2020": {
        "1": {
          "imgurl": "",
          "title": "",
          "description": "",
          "analysis": {
            "HealthIndex": "",
            "Other": ""
          },
          "dayLists":{
            "1": {
              "isUsed": "",
              "HealthIndex" : ""
            }
          }
        }
      },
      "2021": ...
    }
    */
    var res = '{ "data" : ['+
                '{ '+
                  '"year": "2020",'+
                  '"month": ['+
                    '{ "m": "1",' +
                      '"imgurl": "/static/imgs/reports/riqi1.png",'+
                      '"title": "1月报告",'+
                      '"description": "1月报告，良好",'+
                      '"analysis": {'+
                        '"HealthIndex": "",'+
                        '"Other": ""'+
                      '},'+
                      '"dayLists":['+
                        '{ "d" : "31",'+
                          '"isUsed": "",'+
                          '"HealthIndex" : ""'+
                        '}'+
                      ']'+
                    '},'+
                    '{ "m": "2",' +
                      '"imgurl": "/static/imgs/reports/riqi2.png",'+
                      '"title": "2月报告",'+
                      '"description": "2月报告，良好",'+
                      '"analysis": {'+
                        '"HealthIndex": "",'+
                        '"Other": ""'+
                      '},'+
                      '"dayLists":['+
                        '{ "d" : "28",'+
                          '"isUsed": "",'+
                          '"HealthIndex" : ""'+
                        '}'+
                      ']'+
                    '}'+
                  ']'+
                '},'+
                '{ '+
                  '"year": "2021",'+
                  '"month": ['+
                    '{ "m": "2",' +
                      '"imgurl": "/static/imgs/reports/riqi2.png",'+
                      '"title": "2月报告",'+
                      '"description": "2月报告，良好",'+
                      '"analysis": {'+
                        '"HealthIndex": "",'+
                        '"Other": ""'+
                      '},'+
                      '"dayLists":['+
                        '{ "d" : "31",'+
                          '"isUsed": "",'+
                          '"HealthIndex" : ""'+
                          '}'+
                        ']'+
                      '}'+
                    ']'+
                '}'+               
              ']}';
    var obj = JSON.parse(res);
    console.log(obj);
    that.setData({
      reports: obj
    });
  }
})

File: HeartDataServer/src/main/java/com/qian/algorithm/Report.java
package com.qian.algorithm;


import com.alibaba.fastjson.JSONObject;
import com.qian.Entity.algorithm.MonthReport;
import com.qian.utils.Constants;


public class Report {

	/*	生成月报告简略JSON串
	 * 		  "m": "1",
	 * 		  "imgurl": "/static/imgs/reports/riqi1.png", 
	 * 		  "title": "1月报告", 
	 * 	  	  "description": "1月报告，良好", 
	 * 		  "analysis": { 
	 * 			"HealthIndex": "8.0", 
	 * 			"Other": "" 
	 * 		  },
       		  "dayLists": [
       		  	  {
       		  	  	"d" : "1",
                    "imgurl": "/static/imgs/reports/riqi1.png",
                    "isUsed": "true",
                    "HealthIndex" : "9.9",
                    "fileLists":[
                            {
  								"id": "1",
                              	"startTime": "2020-01-01 15:00:00",
                              	"endTime": "2020-01-01 16:00:00",
                              	"avgBeat": "75"
                            },
                            FileReport2...
                     ]
	       		  }, 
	       		  DayReport2...
       		  ]
	 */
	public static JSONObject toReportJSON(MonthReport monthReport){
		JSONObject month = new JSONObject();
		month.put(Constants.M, monthReport.getM());
		month.put(Constants.IMGURL, monthReport.getImgurl());
		month.put(Constants.TITLE, monthReport.getTitle());
		month.put(Constants.DESCRIPTION, monthReport.getDescription());
		
		JSONObject analysis = new JSONObject();
		analysis.put(Constants.HEALTHINDEX, monthReport.healthIndex);
		analysis.put(Constants.OTHER, monthReport.other);
		month.put(Constants.ANALYSIS, analysis);
	
		month.put(Constants.DAYLISTS, monthReport.getDayLists());
		return month;
	}
}
